packages/flutter_tools/lib/src/commands/fuchsia_reload.dart

```diff
@@ -0,0 +1,225 @@
+    _fuchsiaProjectPath = "$_fuchsiaRoot/$_projectRoot";
+    if (!_directoryExists(_fuchsiaProjectPath)) {
+      throwToolExit(
+          "Target does not exist in the Fuchsia tree: $_fuchsiaProjectPath");
+    }
+
+    String relativeTarget = argResults['target'];
+    if (relativeTarget == null) {
+      throwToolExit('Give the application entry point with --target');
+    }
+    _target = "$_fuchsiaProjectPath/$relativeTarget";
+    if (!_fileExists(_target)) {
+      throwToolExit("Couldn't find application entry point at $_target");
+    }
+
+    String buildType = argResults['build-type'];
+    if (buildType == null) {
+      throwToolExit("Give the build type with --build-type");
+    }
+    String packagesFileName = "${_projectName}_dart_package.packages";
+    _dotPackagesPath =
+        "$_fuchsiaRoot/out/$buildType/gen/$_projectRoot/$packagesFileName";
+    if (!_fileExists(_dotPackagesPath)) {
+      throwToolExit("Couldn't find .packages file at $_dotPackagesPath");
+    }
+  }
+
+  List<String> _extractPathAndName(String gnTarget) {
+    String errorMessage =
+        "fuchsia_reload --target '$gnTarget' should have the form: "
+        "'//path/to/app:name'";
+    // Separate strings like //path/to/target:app into [path/to/target, app]
+    int lastColon = gnTarget.lastIndexOf(':');
+    if (lastColon < 0) {
+      throwToolExit(errorMessage);
+    }
+    String name = gnTarget.substring(lastColon + 1);
+    // Skip '//' and chop off after :
+    if ((gnTarget.length < 3) || (gnTarget[0] != '/') || (gnTarget[1] != '/')) {
+      throwToolExit(errorMessage);
+    }
+    String path = gnTarget.substring(2, lastColon);
+    return <String>[path, name];
+  }
+
+  Future<List<int>> _getServicePorts() async {
+    FuchsiaCommandRunner runner = new FuchsiaCommandRunner(_fuchsiaRoot);
+    List<String> lsOutput = await runner.run("ls /tmp/dart.services");
+    List<int> ports = new List<int>();
+    for (String s in lsOutput) {
+      String trimmed = s.trim();
+      int lastSpace = trimmed.lastIndexOf(' ');
+      String lastWord = trimmed.substring(lastSpace + 1);
+      if ((lastWord != '.') && (lastWord != '..')) {
+        int value = int.parse(lastWord, onError: (_) => null);
+        if (value != null) {
+          ports.add(value);
+        }
+      }
+    }
+    return ports;
+  }
+
+  bool _directoryExists(String path) {
+    Directory d = new Directory(path);
+    return d.existsSync();
+  }
+
+  bool _fileExists(String path) {
+    File f = new File(path);
+    return f.existsSync();
+  }
+}
+
+
+// TODO(zra): When Fuchsia has ssh, this should be changed to use that instead.
+class FuchsiaCommandRunner {
+  final String _fuchsiaRoot;
+  final Random _rng = new Random(new DateTime.now().millisecondsSinceEpoch);
+
+  FuchsiaCommandRunner(this._fuchsiaRoot);
+
+  Future<List<String>> run(String command) async {
+    int tag = _rng.nextInt(999999);
+    String netruncmd = "$_fuchsiaRoot/out/build-magenta/tools/netruncmd";
```

Comment:
here and below:

kNetRunCommand = 'out/build-magenta/tools/netruncmd';

fs.path.join(_fuchsiaRoot, kNetRunCommand).